src/main/cpp/src/json_parser.cuh

```diff
@@ -567,6 +508,116 @@ class json_parser {
     }
   }
 
+  __device__ inline std::pair<int, int> try_parse_quoted_string_size(char const* str_pos,
+                                                                     char const quote_char,
+                                                                     char* copy_destination,
+                                                                     escape_style w_style)
+  {
+    // Records string/field name token utf8 bytes size after unescaped
+    // e.g.: For JSON 4 chars string "\\n", after unescaped, get 1 char '\n'
+    // used by checking the max string length
+    int string_token_utf8_bytes = 0;
+    // Records bytes diff between escape writing and unescape writing
+    // e.g.: 4 chars string "\\n", string_token_utf8_bytes is 1,
+    // when `write_escaped_text`, will write out 4 chars: " \ n ",
+    // then this diff will be 4 - 1 = 3
+    int bytes_diff_for_escape_writing = 0;
```

Comment:
Rename to: escped_string_utf8_bytes?
Refactor to record actual bytes instead of diff? will make code more readable.

Reply:
done